File: solutions/LastDigit/solve/java/LastDigitSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class LastDigitSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        long S = (Long) reader.next(Long.class);
        reader.close();

        LastDigit solver = new LastDigit();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.findX(S));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}

File: solutions/LastDigit/solve/cpp/LastDigitSolver.cpp
#include <fstream>
#include <topcoder>
#include "LastDigit.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int64 S; tc::read(ifs, S);
    ifs.close();

    std::ofstream ofs(argv[2]);
    LastDigit solver;
    tc::show(ofs, solver.findX(S));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
